src/ctapipe/monitoring/aggregator.py

```diff
@@ -1,163 +1,465 @@
 """
-Algorithms to compute aggregated time-series statistics from columns of an event table.
+Algorithms to compute aggregated time-series statistics from columns of an astropy table.
 
-These classes take as input an events table, divide it into time chunks, which
-may optionally overlap, and compute various aggregated statistics for each
-chunk. The statistics include the count, mean, median, and standard deviation. The result
-is a monitoring table with columns describing the start and stop time of the chunk
-and the aggregated statistic values.
+These classes take as input an events table containing any event-wise quantities
+(e.g., images, scalars, vectors), divide it into time chunks, which may optionally
+overlap, and compute various aggregated statistics for each chunk. The statistics
+include the count, mean, median, and standard deviation. The result is a monitoring
+table with columns describing the start and stop time of the chunk and the
+aggregated statistic values.
+
+The aggregation is always performed along axis=0 (the event dimension), making
+these classes suitable for any N-dimensional event-wise data.
 """
 
 __all__ = [
+    "BaseChunking",
+    "SizeChunking",
+    "TimeChunking",
+    "BaseAggregator",
     "StatisticsAggregator",
     "PlainAggregator",
     "SigmaClippingAggregator",
 ]
 
-from abc import abstractmethod
+from abc import ABC, abstractmethod
 from collections import defaultdict
+from collections.abc import Generator
 
+import astropy.units as u
 import numpy as np
 from astropy.stats import sigma_clip
 from astropy.table import Table
 
 from ..containers import ChunkStatisticsContainer
-from ..core import TelescopeComponent
-from ..core.traits import Int
-
-__all__ = [
-    "StatisticsAggregator",
-    "PlainAggregator",
-    "SigmaClippingAggregator",
-]
+from ..core import Component
+from ..core.traits import AstroQuantity, Bool, ComponentName, Enum, Int
 
 
-class StatisticsAggregator(TelescopeComponent):
+class BaseChunking(Component, ABC):
```

Comment:
I'm not exactly sure about the implications, but usually we use:

```suggestion
class BaseChunking(Component, metaclass=ABCMeta):
```

Reply:
Oldschool :) class `ABC` was introduced in 3.4 for convenience, there's no implication: 
https://github.com/python/cpython/blob/19ba59369ef55de3ebea721b9c9926c53532d840/Lib/abc.py#L184

I will do as in other places, for uniformity.